File: front/src/app/models/Agent.ts
import {Language} from "./Language";


export class Agent{
  email?:String
  phone?:string
  password?:string
  name?:string

  constructor(email?: string, phone?: string, password?: string,name?:string) {
    this.email = email;
    this.phone = phone;
    this.password = password;
    this.name = name;
  }

}

File: Agentapi/src/main/java/com/botola/agentapi/entities/Agent.java
package com.botola.agentapi.entities;


import jakarta.persistence.*;
import lombok.*;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class Agent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String email;
    private String password;
    private String phone;
    private String language;

    @OneToOne(mappedBy = "agent")
    private Otp otp;

}
